### PR TITLE
Add parse step for wa/state. Fixes #499

### LINKED ISSUE #499: parse wa/state
[learn our pipeline: parse](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#parse)

Blocked by #479

Parse existing data into [`.ndjson`](http://ndjson.org/).

Read all files in the directory passed as the second argument (`sys.argv[2]`), convert them to `.ndjson`, and then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the parse stage and how to get set up for development!

### Tips

1. Fetch data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch wa/state
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest parse wa/state
    ```

### Example
[Fetched files for `md/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/md/arcgis/raw) are converted to [parsed files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/md/arcgis/parsed)

### PR DESCRIPTION
**Pull request by Satshabad**
# Parse State for WA

| Key Details |
|-|
| Resolves #499 |
State: WA |
Site: State |

## Notes
Pretty simple because fetch is json. Just get's the locations and newline's them into `.ndjson`

## Data sample

```json
{"locationId": "recT5VYLb0qwyVTSz", "locationName": "North Basin - Davenport", "locationType": null, "providerId": null, "providerName": null, "departmentId": null, "departmentName": null, "addressLine1": "100 3rd Street", "addressLine2": null, "city": "Davenport", "state": "Washington", "county": "Lincoln", "zipcode": "99122", "latitude": 47.65713, "longitude": -118.14546, "description": null, "contactFirstName": null, "contactLastName": null, "fax": null, "phone": "(509) 725-7501", "email": null, "schedulingLink": "https://www.lincolnhospital.org/health-resources/covid-19-update/", "vaccineAvailability": "AVAILABLE", "infoLink": null, "timeZoneId": null, "directions": "We are currently scheduling weekly vaccine clinics based on the number of vaccines we receive and individual eligibility. The clinics are by appointment only.\n\nPlease call 509.725.7501 and ask to be added to our wait-list. Please note that the vaccine is still available in limited quantities and our demand is higher than our current supply. We will do the best we can to get your scheduled as quickly as possible.", "updatedAt": "2021-05-08T21:15:02.673Z", "rawDataSourceName": "CovidWaFn", "vaccineTypes": [], "accessibleParking": null, "additionalSupports": null, "commCardAvailable": null, "commCardBrailleAvailable": null, "driveupSite": null, "interpretersAvailable": null, "interpretersDesc": null, "supportUrl": null, "waitingArea": null, "walkupSite": null, "wheelchairAccessible": null, "scheduleOnline": null, "scheduleByPhone": null, "scheduleByEmail": null, "walkIn": null, "waitList": null}
{"locationId": "recmLmhezm8iIK6t0", "locationName": "Valley View Health Center - Winlock", "locationType": null, "providerId": null, "providerName": null, "departmentId": null, "departmentName": null, "addressLine1": "100 Cedar Crest Drive", "addressLine2": null, "city": "Winlock", "state": "Washington", "county": "Lewis", "zipcode": "98596", "latitude": 46.49009, "longitude": -122.92417, "description": null, "contactFirstName": null, "contactLastName": null, "fax": null, "phone": "(360) 330-9595", "email": "Covidphase@vvhc.org ", "schedulingLink": "https://vvhc.org/", "vaccineAvailability": "AVAILABLE", "infoLink": null, "timeZoneId": null, "directions": "Waitlist on website", "updatedAt": "2021-05-08T21:15:02.673Z", "rawDataSourceName": "CovidWaFn", "vaccineTypes": [], "accessibleParking": null, "additionalSupports": null, "commCardAvailable": null, "commCardBrailleAvailable": null, "driveupSite": null, "interpretersAvailable": null, "interpretersDesc": null, "supportUrl": null, "waitingArea": null, "walkupSite": null, "wheelchairAccessible": null, "scheduleOnline": null, "scheduleByPhone": null, "scheduleByEmail": null, "walkIn": null, "waitList": null}

```

## Before Opening a PR
- [x] I tested this using the CLI (e.g., `poetry run vaccine-feed-ingest <state>/<site>`)
- [x] I ran auto-formatting: `poetry run tox -e lint-fix`

### CODE REVIEW

**Comment by Satshabad** [comment]
Blocked on #479

File: vaccine_feed_ingest/runners/wa/state/parse.py
Comment (bryanculbertson) [inline vaccine_feed_ingest/runners/wa/state/parse.py:30]: minor: Instead of checking for a `KeyError` can you check each field and then log the error for that field. That will make this easier to fix later if this breaks.

I know this is more verbose, but it definitely will hep with debugging later.

Example:
```
if not data:
    log.error("No data!")
    continue
   
if not data.get("data"):
    log.error("No data data!")
    continue

if not data["data"].get("searchLocations"):
    log.error("No data data searchLocations!")
    continue
    
if not data["data"]["searchLocations"]["locations"]:
    log.warning("No data data searchLocations locations!")
    continue
```

File: vaccine_feed_ingest/runners/wa/state/parse.py
Comment (bryanculbertson) [inline vaccine_feed_ingest/runners/wa/state/parse.py:32]: If there is an error then `locations` will be undefined and this will break.

Can you add a `continue` statement on error so then we can parse the next file?

**Review by bryanculbertson — CHANGES_REQUESTED** [review]
Thank you for this! One minor thing which you can ignore if you don't have time, and one major thing when I would like you to fix before I merge.

File: vaccine_feed_ingest/runners/wa/state/parse.py
Comment (Satshabad) [reply vaccine_feed_ingest/runners/wa/state/parse.py:30]: That's great, thanks. Done

File: vaccine_feed_ingest/runners/wa/state/parse.py
Comment (Satshabad) [reply vaccine_feed_ingest/runners/wa/state/parse.py:32]: done

**Review by bryanculbertson — APPROVED** [review]

**Comment by bryanculbertson** [comment]
@Satshabad Ha! I was expecting you to improve on my quick log statements, but this works :) Approved and merged.

**Pull request merged**
